File: src/main/java/kurenai/imsyncbot/domain/FileCache.java
package kurenai.imsyncbot.domain;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.ColumnDefault;

/**
 * @author Kurenai
 * @since 2023/7/22 19:30
 */


@Entity
@Table(name = "FILE_CACHE")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class FileCache {

    @Id
    private String id;
    private String fileId;
    private String fileType;
    @Version
    @Column(nullable = false)
    @ColumnDefault("0")
    private Integer version;
}

File: src/main/kotlin/kurenai/imsyncbot/domain/FileCache.kt
package kurenai.imsyncbot.domain

import org.babyfish.jimmer.sql.*

/**
 * @author Kurenai
 * @since 2023/7/22 19:30
 */
@Entity
@Table(name = "FILE_CACHE")
interface FileCache {
    @Id
    val id: String
    val fileId: String
    val fileType: String

    @Version
    val version: Int
}